### ISSUE TITLE
Problem with legend when a theme has both legal and projected values

### ISSUE THREAD

**Issue by remyguillaume** [openoereb/pyramid_oereb#1569]
Hello everyone,

Our theme "Nutzungsplanung" has both "legal" (Rechtkräftig) and "projected" (Änderung ohen Vorwirkung) values.
When we generate a PDF Report, 2 pages will be generated : one for the "legal" part, and another one for the "projected" part. That's perfect.

But in the legend, we can see on both pages the both legend entries for "legal" and "projected" theme.
For exemple, this is the "legal" page:

![image](https://user-images.githubusercontent.com/30490367/171565093-c3151239-53f7-4a0f-9c03-e1dbcf53a663.png)

And this is the "projected" page:

![image](https://user-images.githubusercontent.com/30490367/171565141-f4e87ca3-5e6a-4047-a36a-08454c66a61d.png)

On both pages, we can see in the "übreige Legende" part (other legend), that legend-entries are listed, but not visible on the map. It's more obvious on the second page. That's because all legend entries of the theme are loaded, including the "legal" legends, even if there are not any legal object visible in the map.

Actually, for the "legal" page, only the legend entries for the "legal" objects should be displayed, and for the "projected" page, only the legend entries for the "projected" object should be displayed.

Ca we do something here?
Thanks!

**Comment by jwkaltz** [comment]
Hi @remyguillaume ,
can you check if this is already the case in the extract data itself (XML/JSON), or only in the PDF output?

**Comment by remyguillaume** [comment]
Hi @jwkaltz 

This is also the case in the json/xml extracts.

**Comment by voisardf** [comment]
Hi @jwkaltz @remyguillaume @vvmruder 
I can confirm the issue, we have the same problem with 'inKraft' and 'AenderungenMitVorwirkung'.

**Comment by voisardf** [comment]
@remyguillaume  by the way, are you using different WMS services for the different legal states?

**Comment by remyguillaume** [comment]
@voisardf Yes, we use different WMS layers for different legal states.

**Comment by voisardf** [comment]
It seems to me, that this is a regression (or an undiscovered bug?). And yes, this is an important issue.
@michmuel As I heard you're system has passed the exam, so is there a version of pyramid_oereb, where this error was not present yet?

**Comment by michmuel** [comment]
I guess we did not think about the dependency of the legend on the legal status during development.
We passed the exam as we have no other law status than "inForce" in cantonal themes so far. And the transparent symbols in the test data set of the airports security zone plans prevented the pdf extract, while the display in the client was fine.
But we will have other law stati in foreseeable future, so the legend subject should be fixed.
